pyresample/gradient/__init__.py

```diff
@@ -53,258 +53,12 @@ def GradientSearchResampler(source_geo_def, target_geo_def):
 
 def create_gradient_search_resampler(source_geo_def, target_geo_def):
     """Create a gradient search resampler."""
-    if isinstance(source_geo_def, AreaDefinition) and isinstance(target_geo_def, AreaDefinition):
+    if ((isinstance(source_geo_def, AreaDefinition) and isinstance(target_geo_def, AreaDefinition)) or
+        (isinstance(source_geo_def, SwathDefinition) and isinstance(target_geo_def, AreaDefinition))):
```

Comment:
Could these be `is_area_to_swath(source_geo_def, target_geo_def)` and `is_swath_to_area(source_geo_def, target_geo_def)`? Also, this bit seems to be untested, likely also originally.

Reply:
Fixed in [`4dd2948` (#626)](https://github.com/pytroll/pyresample/pull/626/commits/4dd294877c9c24a69890d7634cefeb1f7c9c0270)